test/cmd/apply.sh

```diff
@@ -27,13 +27,13 @@ run_kubectl_apply_tests() {
   kube::log::status "Testing kubectl apply"
   ## kubectl apply should create the resource that doesn't exist yet
   # Pre-Condition: no POD exists
-  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
+  kube::test::get_object_assert pods "{{range.items}}{{${id_field:?}}}:{{end}}" ''
```

Comment:
why :? instead of just the braces fix?

Reply:
Got https://github.com/koalaman/shellcheck/wiki/SC2154.
And from the wiki, it says:

> Note: This message only triggers for variables with lowercase characters in their name (foo and kFOO but not FOO) due to the standard convention of using lowercase variable names for unexported, local variables.

And:

> If you know for a fact that the variable is set, you can use `${var:?}` to fail if the variable is unset (or empty), or explicitly initialize/declare it with `var=""` or `declare var`. You can also disable the message with a directive.

I've tried changing variable name to uppercase and the `${var:?}`  way.
Not sure which is more suitable.

Reply:
this change is reasonable I suppose, but we are not doing this elsewhere 🤔

Reply:
@BenTheElder FYI: https://github.com/kubernetes/kubernetes/pull/73518 is the uppercase way.